providers/implementations/exchange/dh_exch.c

```diff
@@ -102,7 +110,10 @@ static int dh_set_peer(void *vpdhctx, void *vdh)
 {
     PROV_DH_CTX *pdhctx = (PROV_DH_CTX *)vpdhctx;
 
-    if (pdhctx == NULL || vdh == NULL || !DH_up_ref(vdh))
+    if (!ossl_prov_is_running()
```

Comment:
Is this one needed?

Reply:
I think so.  The set_peer calls are doing an up_ref which is equivalent to a new.
There are similar for the other dh exchange algorithms.

I agree that this would probably be a fairly harmless one to let through, but best to block it I think.